cloud/docker.go

```diff
@@ -93,11 +94,10 @@ func (*dockerManager) GetSettings() ProviderSettings {
 
 // SpawnHost creates and starts a new Docker container
 func (m *dockerManager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, error) {
-	if h.Distro.Provider != evergreen.ProviderNameDocker {
+	if h.Distro.Provider != evergreen.ProviderNameDocker && h.Distro.Provider != evergreen.ProviderNameDockerDynamic && h.Distro.Provider != evergreen.ProviderNameDockerStatic {
```

Comment:
I think that dynamic/static are basically all the same so either here or later we'll want to collapse those distinctions (as they're captured entirely in the static provider)